### PR TITLE
Update rails to version 5.2, and misc version bumps to get there

### PR DESCRIPTION
**Pull request by mjankowski**
Summary:

- Rails updated from 5.1 to 5.2
- Misc config/ and bin/ changes as result of running `app:update` - I don't believe any of these actually changed any config options, they just shuffled the files in line with latest rails format/style
- Misc gem version bumps prior to the rails update
- Added bootsnap gem as part of 5.2 update

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -29,4 +97,3 @@
     "Cache-Control" => "public, max-age=31557600",
   }
 end
-Rack::Timeout.timeout = ENV.fetch("RACK_TIMEOUT", 10).to_i
```
Comment (mjankowski) [inline config/environments/production.rb]: This removal is on purpose - the updated version (0.5.1) of rack-timeout [behaves this way by default](https://github.com/heroku/rack-timeout#rails-apps) and doesn't need the setting in place.

File: bin/bundle [hunk 2]
```diff
@@ -1,3 +1,3 @@
 #!/usr/bin/env ruby
-ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../../Gemfile', __FILE__)
+ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../Gemfile', __dir__)
```
Comment (SViccari) [inline bin/bundle]: 🌈 TIL: `__dir__`
> Returns the absolute path of the directory of the file from which this method is called.

File: bin/update [hunk 3]
```diff
@@ -18,6 +17,9 @@ chdir APP_ROOT do
   system! 'gem install bundler --conservative'
   system('bundle check') || system!('bundle install')
 
+  # Install JavaScript dependencies if using Yarn
+  # system('bin/yarn')
```
Comment (SViccari) [inline bin/update]: Does this comment serve as a friendly reminder of how to install JS dependencies via yarn, should we decide to add it in the future?

File: config/cable.yml [hunk 4]
```diff
@@ -6,5 +6,5 @@ test:
 
 production:
   adapter: redis
-  url: redis://localhost:6379/1
-  channel_prefix: bostonrb-website_production
+  url: <%= ENV.fetch("REDIS_URL") { "redis://localhost:6379/1" } %>
```
Comment (SViccari) [inline config/cable.yml]: Cool - I've never passed a block to `ENV.fetch`. It doesn't look like we're using the return value in the block. What do you think of setting this as a default value instead? (ie: `ENV.fetch("REDIS_URL", "redis://localhost:6379/1"`)

File: bin/update (hunk 3 above)
Comment (mjankowski) [reply bin/update]: Yeah - I guess rails is assuming here that you are not using it, but giving you a suggestion of where to run it if you are.

File: config/cable.yml (hunk 4 above)
Comment (mjankowski) [reply config/cable.yml]: I just went with the rails-generated file here ... I'd be fine with either style. At a quick glance, it looks like the rails update generator is just always going with the block style even when there's just a static value to return (puma config has similar stuff).

File: config/initializers/content_security_policy.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Be sure to restart your server when you modify this file.
```
Comment (SViccari) [inline config/initializers/content_security_policy.rb]: This is interesting, did the rails upgrade generate this file or did something else?

File: config/cable.yml (hunk 4 above)
Comment (SViccari) [reply config/cable.yml]: I just noticed that. Keeping it consistent works for me 👍

File: config/puma.rb [hunk 6]
```diff
@@ -1,25 +1,37 @@
...
-# The environment variable WEB_CONCURRENCY may be set to a default value based
-# on dyno size. To manually configure this value use heroku config:set
-# WEB_CONCURRENCY.
+# Specifies the `port` that Puma will listen on to receive requests; default is 3000.
 #
-# Increasing the number of workers will increase the amount of resting memory
-# your dynos use. Increasing the number of threads will increase the amount of
-# potential bloat added to your dynos when they are responding to heavy
-# requests.
+port        ENV.fetch("PORT") { 3000 }
```
Comment (SViccari) [inline config/puma.rb]: So many empty spaces 👾 👾 . Seems odd 🤔

File: config/initializers/content_security_policy.rb (hunk 5 above)
Comment (mjankowski) [reply config/initializers/content_security_policy.rb]: Yes, rails update command generated it ... I usually delete the stuff in config/initializers that's not currently relevant to the project, but this one I suspect will become relevant post-6.0 and/or as apps start to lock down security better.

File: config/puma.rb (hunk 6 above)
Comment (mjankowski) [reply config/puma.rb]: Yes, I cannot endorse this particular generated line's styling.

**Comment by mjankowski** [comment]
FYI - there's now a CVE (loofah) fix included in this PR.

**Pull request merged**